lib/srv/app/cloud.go

```diff
@@ -220,15 +249,28 @@ func (c *cloud) getAWSSigninToken(ctx context.Context, req *AWSSigninRequest, en
 		assumeRole.Duration = duration
 	}
 
+	awsConfigOptions := append(c.cfg.AWSConfigOptions,
+		awsconfig.WithCredentialsMaybeIntegration(integrationMetadata),
+		awsconfig.WithBaseCredentialsProvider(baseCfg.Credentials),
+	)
+
+	// Most flows for providing AWS Access, require the following:
+	// - access to credentials for a helper Role (EC2 instance profile's Role, AWS OIDC Integration Role, etc.)
+	// - use the credentials to call sts:AssumeRole to obtain the credentials for the target role
+	// - use the credentials to call the federation endpoint to obtain the federation URL
+	//
+	// The exception is the IAM Roles Anywhere integration which only uses the target role (no intermediate role).
+	switch {
```

Comment:
Suggestion: What do you think about extracting the detailed assuming role parts to a separate function, and having it be called under this condition? It appears that `temporarySession` and `credentials` are never used in the roles anywhere flow.

```go
	switch {
	case req.RolesAnywhereMetadata.ProfileARN != "":
	default:
		extraOption, err := c.getAssumeDetailedRolesOption(credentials)
		if err != nil {
			return "", trace.Wrap(err)
		}
		awsConfigOptions = append(awsConfigOptions, extraOption)
	}
```